process/block/shardblock_request_test.go

```diff
@@ -0,0 +1,586 @@
+package block_test
+
+import (
+	"bytes"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/stretchr/testify/require"
+
+	blproc "github.com/multiversx/mx-chain-go/process/block"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	dataRetrieverMock "github.com/multiversx/mx-chain-go/testscommon/dataRetriever"
+)
+
+type headerData struct {
+	hash   []byte
+	header data.HeaderHandler
+}
+
+type shardBlockTestData struct {
+	headerData []*headerData
+}
+
+func TestShardProcessor_RequestMissingFinalityAttestingHeaders(t *testing.T) {
+	t.Parallel()
+
+	t.Run("missing attesting meta header", func(t *testing.T) {
+		t.Parallel()
+
+		arguments, requestHandler := shardBlockRequestTestInit(t)
+		testData := createShardProcessorTestData()
+		metaChainData := testData[core.MetachainShardId]
+		numCalls := atomic.Uint32{}
+		requestHandler.RequestShardHeaderByNonceCalled = func(shardID uint32, nonce uint64) {
+			require.Fail(t, fmt.Sprintf("should not request shard header by nonce, shardID: %d, nonce: %d", shardID, nonce))
+		}
+		requestHandler.RequestMetaHeaderByNonceCalled = func(nonce uint64) {
+			attestationNonce := metaChainData.headerData[1].header.GetNonce()
+			if nonce != attestationNonce {
```

Comment:
also here: change to a require.Equal instead?

Reply:
done